sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -700,7 +701,14 @@ case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expressio
       m.region(position, source.length)
       result.delete(0, result.length())
       while (m.find) {
-        m.appendReplacement(result, lastReplacement)
+        try {
+          m.appendReplacement(result, lastReplacement)
+        } catch {
+          case e: Exception =>
+            throw new SparkException(s"Could not perform regexp_replace for " +
```

Comment:
Please, introduce an error condition (class) for the error.

Reply:
Thanks for the suggestion, I have incorporated it